components/PresentorHTMLLayout.js

```diff
@@ -64,6 +64,8 @@ class PresentorHTMLLayout extends React.Component {
                 <script src="/custom_modules/custom-semantic-ui/dist/components/search.min.js" defer></script>
                 <script src="/custom_modules/custom-semantic-ui/dist/components/api.min.js" defer></script>
                 <script src="/custom_modules/custom-semantic-ui/dist/components/sidebar.min.js" defer></script>
+                <script src="/custom_modules/nvd3/d3.v3.min.js" defer></script>
+                <script src="/custom_modules/nvd3/nv.d3.min.js" defer></script>
```

Comment:
First of all, these files don't seem to be included in the `custom_modules` directory. Second, as they appear to be unchanged (by us) libraries which I am sure have npm packages, they should be imported via npm, and not copied over to the `custom_modules`. Haven't tested anything yet however.

Reply:
Hi @kprist, I didn't see they weren't added in custom_modules, now they are, the thing is that they can be downloaded via NPM, however, they must be added as a script as explained here: https://www.npmjs.com/package/nvd3 , so I preferred to add these files in the custom modules folder. In any case, you can check it working on http://localhost:3000/deck/4135-1/a

Reply:
We are using webpack so that we can use the npm packages (and also upgrade them via npm), and also include them in the page as in the directions. There are many ways to do that, for example this: https://stackoverflow.com/questions/31616235/using-d3-js-as-an-external-in-webpack.

But there are also more manual ways to do that, and that's what we're mostly using. Consider these lines of code:
https://github.com/slidewiki/slidewiki-platform/blob/e0e94f1ebb040c2baca6aac0d68e92deb65bc313/components/DefaultHTMLLayout.js#L26-L28

In combination with this: 
https://github.com/slidewiki/slidewiki-platform/blob/e0e94f1ebb040c2baca6aac0d68e92deb65bc313/server.js#L61

You can always ask around for tips on how to include client libraries. The `custom_modules` folder should be used only when we want to use something that we also need to patch, or don't have npm packages (currently only reveal, semantic-ui and some ckeditor plugins that don't even have an npm).

Reply:
@kprist I got it, I didn't know about that configuration on webpack! It's already done. :smiley: